### PR TITLE
chore: add about section & usage examples to access-client readme

### LINKED ISSUE #251: review and update access-client README

### PR DESCRIPTION
**Pull request by yusefnapora**
This adds an About section with some context about the access-client and how it fits into the bigger picture.

It also adds some detail to the Usage section, which now has subsections for creating Agents & Spaces and registering Spaces with the access service.

Closes #251

### CODE REVIEW

File: packages/access-client/readme.md [hunk 1]
```diff
@@ -28,6 +51,78 @@ if (!(await store.exists())) {
...
+```
+
+Note that the receiving agent will need to [import the delegation](#importing-delegations-from-another-agent) before they will be able to invoke the delegated capabilities.
+
+### Importing delegations from another Agent
+
+The [`addProof` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#addProof) takes in a ucanto `Delegation` and adds it to the Agent's state Store. The proof of delegation can be retrieved using the Agent's [`proofs` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#proofs).
+
+
+The [`importSpaceFromDelegation` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#importSpaceFromDelegation) also accepts a ucanto `Delegation`, but it is tailored for "full delegation" of all Space-related capabilities. The delegated ability must be `space/*`, which is the "top" of the `space/` ability set. Use `importSpaceFromDelegation` in preference to `addProofs` when importing a full `space/*` delegation, as it also adds metadata about the imported Space to the Agent's `spaces` Map and persistent Store.
```
Comment (yusefnapora) [inline packages/access-client/readme.md:124]: This link doesn't resolve at the moment, but it should once the type issues that are preventing typedoc from doing its thing are resolved.

**Comment by yusefnapora** [comment]
Turns out my type errors were caused by not running `pnpm run build` before `pnpm run docs`. Made a tiny PR to always build first: #266

File: packages/access-client/readme.md (hunk 1 above)
Comment (alanshaw) [inline packages/access-client/readme.md:124]: The delegated ability must be "*" (top). I'm not sure why this is restricted right now @hugomrdias?

I'd avoid talking about the internal data structure (`Map`) here also, but obviously mention that it means the agent tracks it as a "space".

**Review by alanshaw — APPROVED** [review]
This LGTM - please address the feedback and feel free to merge.

File: packages/access-client/readme.md [hunk 2]
```diff
@@ -13,13 +23,26 @@ npm install @web3-storage/access
...
 ```js
-import { Agent, connection } from '@web3-storage/access/agent'
-import * as Encoding from '@web3-storage/access/encoding'
-import { StoreConf } from '@web3-storage/access/stores/store-conf'
-// for browsers
+import { Agent } from '@web3-storage/access/agent'
 import { StoreIndexedDB } from '@web3-storage/access/stores/store-indexeddb'
 
+const store = await StoreIndexedDB.open('my-db-name')
+const agent = await Agent.create({ store })
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply packages/access-client/readme.md:37]: The pattern goes like this 

```js
const store = new StoreIndexedDB('my-db-name')
const data = await store.load()
if ( data ){
  const agent = Agent.from(data, { store })
} else {
  await Agent.create({ 
    meta : { name: 'agent-name' },
    principal: await RSASigner.generate(), // principal is optional and EdSigner is the default 
  } , { store }
}
```

File: packages/access-client/readme.md (hunk 1 above)
Comment (hugomrdias) [reply packages/access-client/readme.md:124]: The concepts here are hard to distinguish, meaning delegation vs space. What is a space ? It's a DID plus a delegation to do something with it. But you can have a delegation to just do 'store/add' in a space, does that mean you have a that space? No, you have the just that capability. So for now until we have a better way to design this.

Space === `{ can: '*', with: did:space, audience: did:agent }` and Proof === `{ can: 'less than *', with: did:space, audience: did:agent }`

And delegations === `{ can: 'anything agent can delegate', with: did:space, audience: did:someOtherAgent }`

So theres these 3 concepts here, 

1. **_Space_** meaning dids the agent has full capabilities, 
2. **_Proofs_** meaning all the delegations where the audience is the Agent 
3. **_Delegations_** meaning all the delegations the agent create to another agent.

**Review by hugomrdias — CHANGES_REQUESTED** [review]

**Pull request merged**